src/main/java/org/cardanofoundation/cip8/CIP8Parser.java

```diff
@@ -28,25 +28,65 @@
 import static org.cardanofoundation.ext.cbor.MoreCbor.serialize;
 import static org.cardanofoundation.ext.cose.COSEKey.deserialize;
 
+/**
+ * The {@code CIP8Parser} class is used to verify and parse a message that uses the CIP 8
+ * specification of the Cardano protocol.
+ * For further information on CIP 8, see <A href="https://cips.cardano.org/cips/cip8/">
+ * https://cips.cardano.org/cips/cip8/</A>.
+ */
 public final class CIP8Parser {
 
+    /**
+     * Static instance of {@code Logger} used for logging.
+     */
     private static final Logger logger = LoggerFactory.getLogger(CIP8Parser.class);
 
+    /**
+     * TODO
+     */
     private static final EdDSAParameterSpec ED_DSA_PARAMETER_SPEC = EdDSANamedCurveTable.getByName(ED_25519);
+
+    /**
+     * TODO
+     */
     private static final long PUBLIC_KEY_INDEX = -2;
 
+    /**
+     * Signature of the message signed using CIP 8.
+     */
     private final String signature;
+
+    /**
+     * Public key used to sign the message using CIP 8.
+     */
     private final Optional<String> publicKey;
 
+    /**
+     * Creates an {@code CIP8Parser}.
+     *
+     * @param signature a string that contains the signature of the message signed using CIP 8.
```

Comment:
I can sense a bit tautology here, I wonder if we can rewrite this a bit...
```
@param signature is a string that contains the COSE payload with message signed as per CIP-8 specification.
```

Something like this, not sure exactly of wording. I think we didn't talk about glossary and these terms like Cose payload, payload, message they are seem the same but actually we could be very precise what we mean by those terms. We need some GLOSSARY.md document in the repository because I remember when I was talking to PI from SundaeSwap, there was a lot of confusion around many terms and they meaning. We can pair program on this and create such GLOSSARY together.